In [1]:
import pandas as pd
import numpy as np

In [4]:
df = pd.read_csv("data/EMSResponses.csv")
df.head()



incident_ID incident_date  priority            zone    time_phone_pickup  \
0            1      1/1/2012         4  City of Austin  12/31/2011 20:32:54   
1            2      1/1/2012         4  City of Austin  01/01/2012 00:03:46   
2            3      1/1/2012         4  City of Austin  01/01/2012 00:08:43   
3            4      1/1/2012         2  City of Austin  01/01/2012 00:11:47   
4            5      1/1/2012         3  City of Austin  01/01/2012 00:16:14   

     time_unit_arrived time_incident_closed               disposition  \
0  01/01/2012 03:03:07  01/01/2012 03:40:39          ATCEMS Transport   
1  01/01/2012 00:35:56  01/01/2012 01:21:12           Patient Refusal   
2                  NaN  01/01/2012 00:11:29  No Patient / False Alarm   
3  01/01/2012 00:17:22  01/01/2012 01:03:06           Patient Refusal   
4  01/01/2012 00:24:58  01/01/2012 01:09:11          ATCEMS Transport   

   council_district  census_state_code  census_county_code  census_tract  \
0                 9                 48                 453          1100   
1                 5                 48                 453          2428   
2                 9                 48                 453          1100   
3                 3                 48                 453          2314   
4                 2                 48                 453          2430   

   census_block     census_geoID census_block_name  
0          1118  484530011001118        Block 1118  
1          1003  484530024281003        Block 1003  
2          1080  484530011001080        Block 1080  
3          2010  484530023142010        Block 2010  
4          1019  484530024301019        Block 1019

In [5]:
#Remove all of the temporal and non-unique data that we don't care about.
df.drop(["census_state_code", "census_county_code", "census_block_name", "zone"], axis = 1, inplace = True)

In [6]:
df.drop(["incident_date", "time_phone_pickup", "time_unit_arrived","time_incident_closed"], axis = 1, inplace = True)

In [7]:
df["disposition"][df["disposition"] == "<Missing>"] = "Other"


/Users/kevinmagana/anaconda/envs/python3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 620352 entries, 0 to 620351
Data columns (total 7 columns):
incident_ID         620352 non-null int64
priority            620352 non-null int64
disposition         620352 non-null object
council_district    620352 non-null int64
census_tract        620352 non-null int64
census_block        620352 non-null int64
census_geoID        620352 non-null int64
dtypes: int64(6), object(1)
memory usage: 33.1+ MB


In [75]:
df2 = df.groupby("disposition").size().reset_index(name="total_incidents")
df2["average_per_month"] = (df2["total_incidents"] / 58).round()
df2["average_per_year"] = df2["average_per_month"] * 12
df2

disposition  total_incidents  average_per_month  \
0          ATCEMS Transport           381823             6583.0   
1                 Cancelled            43441              749.0   
2         Deceased on Scene             4814               83.0   
3  No Patient / False Alarm            58220             1004.0   
4                     Other             1312               23.0   
5           Patient Refusal            95381             1644.0   
6                  Referred            35332              609.0   
7                      Test               29                0.0   

   average_per_year  
0           78996.0  
1            8988.0  
2             996.0  
3           12048.0  
4             276.0  
5           19728.0  
6            7308.0  
7               0.0

Between 01/01 2012 and 10/31/2017, ATCEMS transport was dispatched to 381823 incidents

# Prevalence Data

In [99]:
df3 = pd.read_csv("data/AdultsArthritis.csv")
df3 = df3[["Location", "Indicator Value"]]
df3=df3.rename(columns = {'Indicator Value':'Arthritis'})

In [104]:
files = ["data/AdultsCancer.csv", "data/AdultsDiabetes.csv", "data/AdultsHighCholesterol.csv", "data/AdultsObese.csv",\
         "data/HighBloodPressure.csv", "data/AdultsRoutineCheckup.csv", "data/AdultsWithoutHealthInsurance.csv",\
         "data/ElderlyBelowPovertyLevel.csv", "data/ElderlyLivingAlone.csv",\
         "data/ElderlyRecievedRecommendedPreventationServicesFemales.csv","data/ElderlyRecievedRecommendedPreventativeServicesMales.csv",\
         "data/HouseholdsWithoutVehicle.csv"]

labels = ["Cancer", "Diabetes", "Cholesterol", "Obese", "HBP", "Checkup", "NoInsurance", "Poverty", "Alone",\
          "FemaleServices", "MaleServices", "NoVehicles"]

df2 = pd.read_csv("data/AdultsCancer.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'Cancer'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/AdultsDiabetes.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'Diabetes'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/AdultsHighCholesterol.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'Cholesterol'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/AdultsObese.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'Obese'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/HighBloodPressure.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'HBP'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/AdultsRoutineCheckup.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'Routine'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/AdultsWithoutHealthInsurance.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'NoInsurance'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/ElderlyBelowPovertyLevel.csv")
df2 = df2[df2["Period of Measure"] == "2011-2015"]
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'ElderlyPoverty'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/ElderlyLivingAlone.csv")
df2 = df2[df2["Period of Measure"] == "2011-2015"]
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'ElderlyAlone'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/ElderlyRecievedRecommendedPreventationServicesFemales.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'FemaleServices'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/ElderlyRecievedRecommendedPreventativeServicesMales.csv")
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'MaleServices'})
df3 = df3.merge(df2, on = "Location")

df2 = pd.read_csv("data/HouseholdsWithoutVehicle.csv")
df2 = df2[df2["Period of Measure"] == "2011-2015"]
df2 = df2[["Location", "Indicator Value"]]
df2=df2.rename(columns = {'Indicator Value':'NoVehicle'})
df3 = df3.merge(df2, on = "Location")

In [101]:
df3

Location  Arthritis  Cancer  Diabetes  Cholesterol  Obese   HBP  \
0    48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
1    48453000102       20.5     8.8       7.8         41.9   19.8  32.0   
2    48453000203        6.9     2.3       3.2         22.6   17.8  14.3   
3    48453000204       12.0     4.1       5.1         30.9   20.9  20.2   
4    48453000205       11.1     3.8       5.1         29.4   21.6  19.3   
5    48453000206       16.8     6.3       7.1         36.2   21.6  26.7   
6    48453000302       10.0     3.4       4.5         28.2   20.0  18.0   
7    48453000304        9.1     2.9       4.5         27.4   21.3  18.2   
8    48453000305        9.4     3.2       4.2         26.8   20.7  17.0   
9    48453000306       10.3     3.3       5.3         28.1   22.6  18.7   
10   48453000307       15.0     4.9       7.2         33.9   24.2  25.0   
11   48453000401        8.3     2.8       3.9         25.4   18.5  16.1   
12   48453000402       14.7     4.6       8.3         31.6   24.0  26.1   
13   48453000500        8.1     2.6       3.7         24.5   19.6  15.2   
14   48453000603        4.3     1.0       2.2         14.9   20.0  11.1   
15   48453000700       12.2     4.5       5.8         33.2   19.4  23.4   
16   48453000801       22.7     5.4      17.5         41.5   32.3  34.8   
17   48453000802       19.3     4.0      14.1         36.1   35.4  33.8   
18   48453000803       13.2     3.6       8.2         30.2   26.5  24.3   
19   48453000804       17.4     4.0      12.2         35.2   32.5  31.1   
20   48453000901       11.5     3.1       7.7         29.8   25.7  22.7   
21   48453000902       16.3     3.7      12.6         37.6   31.2  28.9   
22   48453001000       18.0     4.8      12.7         38.3   28.2  29.1   
23   48453001100       11.4     4.1       5.7         31.3   21.8  21.9   
24   48453001200       10.1     4.1       4.1         29.1   18.4  18.7   
25   48453001303       14.0     5.2       5.8         34.2   21.5  23.5   
26   48453001304       12.6     4.6       5.2         31.3   21.5  21.0   
27   48453001305       12.5     3.9       6.5         31.8   23.9  22.0   
28   48453001307       13.7     4.0       7.8         33.2   25.9  23.5   
29   48453001308       13.9     4.0       8.2         32.4   25.5  22.4   
..           ...        ...     ...       ...          ...    ...   ...   
156  48453002313       10.8     2.6       6.8         28.0   27.6  19.1   
157  48453002314       10.1     2.1       7.0         27.7   29.9  20.3   
158  48453002315       11.0     2.3       8.2         29.0   31.2  20.2   
159  48453002316       11.5     2.1       9.3         30.1   33.6  21.9   
160  48453002317        5.8     1.2       4.3         20.0   24.3  14.6   
161  48453002318       11.2     2.2       8.4         29.0   28.9  20.3   
162  48453002402       12.6     3.4       7.7         31.6   25.6  21.5   
163  48453002403       15.5     4.4       9.2         34.7   26.6  24.0   
164  48453002407       13.4     4.3       6.9         31.7   23.4  20.8   
165  48453002409       17.0     4.4      10.7         36.9   29.3  27.5   
166  48453002410       15.4     3.8       9.8         34.4   29.5  24.7   
167  48453002411       15.8     2.8      13.1         35.1   37.1  26.2   
168  48453002412       15.7     3.1      12.8         34.9   34.2  24.8   
169  48453002413       14.7     2.5      12.2         33.4   37.0  25.5   
170  48453002419       10.1     2.2       7.0         27.7   29.2  19.3   
171  48453002421       10.7     3.3       5.7         28.5   23.3  18.5   
172  48453002422       15.4     4.1       9.4         34.8   28.2  24.7   
173  48453002423       15.3     4.2       8.8         35.1   27.4  25.2   
174  48453002424       17.3     4.7      10.2         36.6   28.5  26.9   
175  48453002425       10.5     3.2       6.0         28.7   23.9  18.7   
176  48453002426       11.7     2.8       8.2         30.0   29.2  20.6   
177  48453002427       15.6     3

In [105]:
df3.to_csv("IndicatorsData.csv")

In [96]:
df2.head()

Indicator Name  \
0  People 65+ Living Below Poverty Level   
1  People 65+ Living Below Poverty Level   
2  People 65+ Living Below Poverty Level   
3  People 65+ Living Below Poverty Level   
4  People 65+ Living Below Poverty Level   

                              What Is This Indicator Location Type  \
0  This indicator shows the percentage of people ...  Census Tract   
1  This indicator shows the percentage of people ...  Census Tract   
2  This indicator shows the percentage of people ...  Census Tract   
3  This indicator shows the percentage of people ...  Census Tract   
4  This indicator shows the percentage of people ...  Census Tract   

      Location  Indicator Value Indicator Value Units  \
0  48453000101              5.1               percent   
1  48453000101              8.8               percent   
2  48453000101             13.4               percent   
3  48453000101             14.5               percent   
4  48453000101             16.4               percent   

   Lower Confidence Interval  Upper Confidence Interval  \
0                       -0.3                       10.5   
1                        1.1                       16.5   
2                        3.0                       23.8   
3                        3.0                       26.0   
4                        NaN                        NaN   

  Indicator Value Unstable Period of Measure       ...        Technical Note  \
0                       No         2011-2015       ...                   NaN   
1                       No         2010-2014       ...                   NaN   
2                       No         2009-2013       ...                   NaN   
3                       No         2008-2012       ...                   NaN   
4                       No         2007-2011       ...                   NaN   

   Breakout Title  Breakout Category  Breakout Subcategory  Breakout Value  \
0             NaN                NaN                   NaN             NaN   
1             NaN                NaN                   NaN             NaN   
2             NaN                NaN                   NaN             NaN   
3             NaN                NaN                   NaN             NaN   
4             NaN                NaN                   NaN             NaN   

   Breakout Value Units  Breakout Lower Confidence Interval  \
0                   NaN                                 NaN   
1                   NaN                                 NaN   
2                   NaN                                 NaN   
3                   NaN                                 NaN   
4                   NaN                                 NaN   

   Breakout Upper Confidence Interval  Breakout Unstable  Breakout Footer  
0                                 NaN                NaN              NaN  
1                                 NaN                NaN              NaN  
2                                 NaN                NaN              NaN  
3                                 NaN                NaN              NaN  
4                                 NaN                NaN              NaN  

[5 rows x 21 columns]

Location  Arthritis  Cancer  Diabetes  Cholesterol  Obese   HBP  \
0      48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
1      48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
2      48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
3      48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
4      48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
5      48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
6      48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
7      48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
8      48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
9      48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
10     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
11     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
12     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
13     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
14     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
15     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
16     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
17     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
18     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
19     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
20     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
21     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
22     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
23     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
24     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
25     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
26     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
27     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
28     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
29     48453000101       17.4     7.3       6.5         35.8   19.3  26.9   
...            ...        ...     ...       ...          ...    ...   ...   
46073  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46074  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46075  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46076  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46077  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46078  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46079  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46080  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46081  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46082  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46083  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46084  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46085  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46086  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46087  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46088  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46089  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46090  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46091  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46092  48453002436        NaN     NaN       NaN          NaN    NaN   NaN   
46093 

In [102]:
df2 = pd.read_csv("data/HouseholdsWithoutVehicle.csv")

In [103]:
df2.head()

Indicator Name  \
0  Households without a Vehicle   
1  Households without a Vehicle   
2  Households without a Vehicle   
3  Households without a Vehicle   
4  Households without a Vehicle   

                              What Is This Indicator Location Type  \
0  This indicator shows the percentage of househo...  Census Tract   
1  This indicator shows the percentage of househo...  Census Tract   
2  This indicator shows the percentage of househo...  Census Tract   
3  This indicator shows the percentage of househo...  Census Tract   
4  This indicator shows the percentage of househo...  Census Tract   

      Location  Indicator Value Indicator Value Units  \
0  48453000101              8.8               percent   
1  48453000101              9.8               percent   
2  48453000101             11.6               percent   
3  48453000101             12.9               percent   
4  48453000101              9.3               percent   

   Lower Confidence Interval  Upper Confidence Interval  \
0                        5.2                       12.4   
1                        5.6                       14.0   
2                        7.1                       16.1   
3                        8.8                       17.0   
4                        NaN                        NaN   

  Indicator Value Unstable Period of Measure       ...        Technical Note  \
0                       No         2011-2015       ...                   NaN   
1                       No         2010-2014       ...                   NaN   
2                       No         2009-2013       ...                   NaN   
3                       No         2008-2012       ...                   NaN   
4                       No         2007-2011       ...                   NaN   

   Breakout Title  Breakout Category  Breakout Subcategory  Breakout Value  \
0             NaN                NaN                   NaN             NaN   
1             NaN                NaN                   NaN             NaN   
2             NaN                NaN                   NaN             NaN   
3             NaN                NaN                   NaN             NaN   
4             NaN                NaN                   NaN             NaN   

   Breakout Value Units  Breakout Lower Confidence Interval  \
0                   NaN                                 NaN   
1                   NaN                                 NaN   
2                   NaN                                 NaN   
3                   NaN                                 NaN   
4                   NaN                                 NaN   

   Breakout Upper Confidence Interval  Breakout Unstable  Breakout Footer  
0                                 NaN                NaN              NaN  
1                                 NaN                NaN              NaN  
2                                 NaN                NaN              NaN  
3                                 NaN                NaN              NaN  
4                                 NaN                NaN              NaN  

[5 rows x 21 columns]